Build failure: python312Packages.onnx

### Steps To Reproduce

Steps to reproduce the behavior:
1. build python312Packages.onnx as a part of immich-machine-learning

### Build log

<details>
<summary>Build log</summary>
-- Download and build Abseil from https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz
[ 11%] Creating directories for 'abseil-populate'
[ 22%] Performing download step (download, verify and extract) for 'abseil-populate'
-- Downloading...
   dst='/build/source/.setuptools-cmake-build/_deps/abseil-subbuild/abseil-populate-prefix/src/20230125.3.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
-- Retrying...
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
-- Retry after 5 seconds (attempt #2) ...
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
-- Retry after 5 seconds (attempt #3) ...
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
-- Retry after 15 seconds (attempt #4) ...
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
-- Retry after 60 seconds (attempt #5) ...
-- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz'
CMake Error at abseil-subbuild/abseil-populate-prefix/src/abseil-populate-stamp/download-abseil-populate.cmake:162 (message):
  Each download failed!

    error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---
          error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---
          error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---
          error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---
          error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---
          error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20230125.3.tar.gz' failed
          status_code: 6
          status_string: "Could not resolve hostname"
          log:
          --- LOG BEGIN ---
          Could not resolve host: github.com

  shutting down connection #0



          --- LOG END ---




make[2]: *** [CMakeFiles/abseil-populate.dir/build.make:100: abseil-populate-prefix/src/abseil-populate-stamp/abseil-populate-download] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/abseil-populate.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

CMake Error at /nix/store/24gbi6mcw4c6jgzai9aaxx9qi2rxdb09-cmake-3.29.6/share/cmake-3.29/Modules/FetchContent.cmake:1714 (message):
  Build step for abseil failed: 2
Call Stack (most recent call first):
  /nix/store/24gbi6mcw4c6jgzai9aaxx9qi2rxdb09-cmake-3.29.6/share/cmake-3.29/Modules/FetchContent.cmake:1854:EVAL:2 (__FetchContent_directPopulate)
  /nix/store/24gbi6mcw4c6jgzai9aaxx9qi2rxdb09-cmake-3.29.6/share/cmake-3.29/Modules/FetchContent.cmake:1854 (cmake_language)
  CMakeLists.txt:250 (FetchContent_Populate)


-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/build/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 309, in <module>
    setuptools.setup(
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/__init__.py", line 108, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 184, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
    dist.run_commands()
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 970, in run_commands
    self.run_command(cmd)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
    super().run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/command/bdist_wheel.py", line 373, in run
    self.run_command("build")
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
    self.distribution.run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
    super().run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
    self.distribution.run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
    super().run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "setup.py", line 246, in run
    self.run_command("cmake_build")
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
    self.distribution.run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/dist.py", line 945, in run_command
    super().run_command(command)
  File "/nix/store/l6vlavpj6wzcihb1qmrh1xh8wh22cmfb-python3.12-setuptools-72.1.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "setup.py", line 217, in run
    subprocess.check_call(cmake_args)
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/slflwdf69f5x21amc4zac5dpa8gk7kc5-python3.12-cmake-3.29.6/bin/cmake', '-DPYTHON_INCLUDE_DIR=/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/include/python3.12', '-DPYTHON_EXECUTABLE=/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/bin/python3.12', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-312-x86_64-linux-gnu.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '-DONNX_BUILD_TESTS=ON', '-DCMAKE_INSTALL_LIBDIR=lib', '-DONNX_USE_PROTOBUF_SHARED_LIBS=ON', '-Dgoogletest_STATIC_LIBRARIES=/nix/store/m8i80frc3qbb0d9ifh3mvfwkd4yvz790-gtest-1.15.2/lib/libgtest.a', '-Dgoogletest_INCLUDE_DIRS=/nix/store/k7kwvbr8l81wiw3fi5sydpfsylqbniw0-gtest-1.15.2-dev/include', '/build/source']' returned non-zero exit status 1.

</details>

### Additional context

This was triggered as dependency of immich-machine-learning, so currently on the unstable branch of nixpkgs, immich is broken.

### Notify maintainers

@acairncross

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.13, NixOS, 24.11 (Vicuna), 24.11.20241009.5633bcf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/60sn02zhawl3kwn0r515zff3h6hg6ydz-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/onnx/default.nix
+++ b/pkgs/development/python-modules/onnx/default.nix
@@ -26,7 +26,7 @@ let
 in
 buildPythonPackage rec {
   pname = "onnx";
-  version = "1.16.2";
+  version = "1.17.0";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -35,7 +35,7 @@ buildPythonPackage rec {
     owner = "onnx";
     repo = "onnx";
     rev = "refs/tags/v${version}";
-    hash = "sha256-JmxnsHRrzj2QzPz3Yndw0MmgZJ8MDYxHjuQ7PQkQsDg=";
+    hash = "sha256-9oORW0YlQ6SphqfbjcYb0dTlHc+1gzy9quH/Lj6By8Q=";
   };
 
   build-system = [
@@ -75,22 +75,13 @@ buildPythonPackage rec {
 
     chmod +x tools/protoc-gen-mypy.sh.in
     patchShebangs tools/protoc-gen-mypy.sh.in
-
-    substituteInPlace setup.py \
-      --replace 'setup_requires.append("pytest-runner")' ""
-
-    # prevent from fetching & building own gtest
-    substituteInPlace CMakeLists.txt \
-      --replace 'include(googletest)' ""
-    substituteInPlace cmake/unittest.cmake \
-      --replace 'googletest)' ')'
   '';
 
   preConfigure = ''
     # Set CMAKE_INSTALL_LIBDIR to lib explicitly, because otherwise it gets set
     # to lib64 and cmake incorrectly looks for the protobuf library in lib64
     export CMAKE_ARGS="-DCMAKE_INSTALL_LIBDIR=lib -DONNX_USE_PROTOBUF_SHARED_LIBS=ON"
-    export CMAKE_ARGS+=" -Dgoogletest_STATIC_LIBRARIES=${gtestStatic}/lib/libgtest.a -Dgoogletest_INCLUDE_DIRS=${lib.getDev gtestStatic}/include"
+    export CMAKE_ARGS+=" -Dgoogletest_STATIC_LIBRARIES=${gtestStatic}/lib/libgtest.a"
     export ONNX_BUILD_TESTS=1
   '';
 
@@ -115,7 +106,7 @@ buildPythonPackage rec {
 
   pytestFlagsArray = [
     "onnx/test"
-    "onnx/examples"
+    "examples"
   ];
 
   disabledTests =
@@ -143,11 +134,6 @@ buildPythonPackage rec {
       "test_ops_tested"
     ];
 
-  disabledTestPaths = [
-    # Unexpected output fields from running code: {'stderr'}
-    "onnx/examples/np_array_tensorproto.ipynb"
-  ];
-
   __darwinAllowLocalNetworking = true;
 
   postCheck = ''
@@ -158,6 +144,7 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "onnx" ];
 
   meta = with lib; {
+    changelog = "https://github.com/onnx/onnx/releases/tag/${lib.removePrefix "refs/tags/" src.rev}";
     description = "Open Neural Network Exchange";
     homepage = "https://onnx.ai";
     license = licenses.asl20;

--- a/pkgs/development/python-modules/onnx/default.nix
+++ b/pkgs/development/python-modules/onnx/default.nix
@@ -18,6 +18,7 @@
   google-re2,
   pillow,
   protobuf,
+  setuptools,
 }:
 
 let
@@ -26,45 +27,52 @@ in
 buildPythonPackage rec {
   pname = "onnx";
   version = "1.16.2";
-  format = "setuptools";
+  pyproject = true;
 
   disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
-    owner = pname;
-    repo = pname;
+    owner = "onnx";
+    repo = "onnx";
     rev = "refs/tags/v${version}";
     hash = "sha256-JmxnsHRrzj2QzPz3Yndw0MmgZJ8MDYxHjuQ7PQkQsDg=";
   };
 
   build-system = [
     cmake
-    pybind11
+    protobuf
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    protobuf_21 # for protoc
   ];
 
   buildInputs = [
     abseil-cpp
-    protobuf
-    google-re2
+    protobuf_21
     gtestStatic
-    pillow
+    pybind11
   ];
 
   dependencies = [
-    protobuf_21
     protobuf
     numpy
     typing-extensions
   ];
 
   nativeCheckInputs = [
+    google-re2
     nbval
     parameterized
+    pillow
     pytestCheckHook
     tabulate
   ];
 
   postPatch = ''
+    rm -r third_party
+
     chmod +x tools/protoc-gen-mypy.sh.in
     patchShebangs tools/protoc-gen-mypy.sh.in
 

